File: week7/partialab.java
abstract class Shape
{
abstract float area();
void display()
{
System.out.println("This is a shape");
}
}
class Rectangle extends Shape
{
float b,h;
Rectangle(float base,float height)
{
b=base;
h=height;
}
public float area()
{
return b*h;
}
}
class Triangle extends Shape
{
float b,h;
Triangle(float base,float height)
{
b=base;
h=height;
}
public float area()
{
return (float)0.50*b*h;
}
}
public class PartialAbstraction
{
public static void main(String[] args)
{
Rectangle obj1=new Rectangle(24,64);
Triangle obj2=new Triangle(30,11);
System.out.println("Area of Rectangle : "+obj1.area());
System.out.println("Area of Triangle : "+obj2.area());
}
}

File: week7/partialab.cpp
#include<iostream>
using namespace std;
class Shape
{
private:
virtual float area()=0;
public:
void displayMSG()
{
cout<<"This is a shape\n";
}
};
class Triangle: public Shape
{
public:
float b,h;
Triangle(float base,float height)
{
b=base;
h=height;
}
float area()
{
return 0.5*b*h;
}
};
class Circle: public Shape
{
public:
float r;
Circle(float radius)
{
r=radius;
}
float area()
{
return 3.14*r*r;
}
};
int main()
{
Triangle obj1(2,4);
Circle obj2(3.0);
obj1.displayMSG();
cout<<"Area of the Triangle : "<<obj1.area()<<endl;
obj2.displayMSG();
cout<<"Area of the Circle: "<<obj2.area()<<endl;
return 0;
}
